### PR TITLE
Remove AzCoreTestCommon, move related files to AzTest library

### PR DESCRIPTION
**Pull request by burelc-amzn**
AzCoreTestCommon includes headers from AzTest, and AzTest requires
symbols that are in AzCoreTestCommon. This change removes that circular
dependency.

Signed-off-by: Chris Burel <burelc@amazon.com>

### CODE REVIEW

File: Code/Framework/AzTest/AzTest/AzTest.h [hunk 1]
```diff
@@ -193,14 +193,9 @@ namespace AZ
...
-            static AZ::EnvironmentVariable<AZ::Test::BenchmarkEnvironmentRegistry> s_benchmarkRegistry;
-            if (!s_benchmarkRegistry)
-            {
-                s_benchmarkRegistry = AZ::Environment::CreateVariable<AZ::Test::BenchmarkEnvironmentRegistry>(s_benchmarkEnvironmentName);
-            }
-
+            static AZ::Test::BenchmarkEnvironmentRegistry s_benchmarkRegistry;
             auto benchmarkEnv{ new T };
-            s_benchmarkRegistry->AddBenchmarkEnvironment(std::unique_ptr<BenchmarkEnvironmentBase>{ benchmarkEnv });
+            s_benchmarkRegistry.AddBenchmarkEnvironment(std::unique_ptr<BenchmarkEnvironmentBase>{ benchmarkEnv });
```
Comment (AMZN-ScottR) [inline Code/Framework/AzTest/AzTest/AzTest.h:198]: This has some behavioural changes that I don't think are intended.  The first of which is the `BenchmarkEnvironmentRegistry` env variable is no longer being created, so the calls to `FindVariable` in the `AZ_BENCHMARK_HOOK*` macros won't process registered environments.

File: Code/Framework/AzTest/AzTest/UnitTest.h [hunk 2]
```diff
@@ -10,12 +10,6 @@
 #include <AzTest/AzTest.h>
```
Comment (Kadino) [inline Code/Framework/AzTest/AzTest/UnitTest.h:10]: I need a bit to mull over this change, particularly in regards to enshrining the UnitTest namespace as an example to other modules.

File: Code/Framework/AzTest/AzTest/aztest_files.cmake [hunk 3]
```diff
@@ -7,15 +7,23 @@
 #
 
 set(FILES
-    AzTest.h
     AzTest.cpp
+    AzTest.h
     ColorizedOutput.cpp
+    GemTestEnvironment.cpp
+    GemTestEnvironment.h
+    MockComponentApplication.cpp
```
Comment (Kadino) [inline Code/Framework/AzTest/AzTest/aztest_files.cmake:15]: AzTest is used by many libraries and does not include mocks that are specific to them. What is the argument for AzCore mocks living inside AzTest? Should they not exist in a separate sister library?

File: Code/Framework/AzTest/AzTest/UnitTest.h (hunk 2 above)
Comment (Kadino) [reply Code/Framework/AzTest/AzTest/UnitTest.h:10]: Interesting, the UnitTest header/namespace had already been injected into AzTest.

Comment (Kadino) [reply Code/Framework/AzTest/AzTest/UnitTest.h:10]: I'm okay with this dependency moving into AzTest, though not most of the other files

File: Code/Framework/AzTest/AzTest/aztest_files.cmake [hunk 4]
```diff
@@ -7,15 +7,23 @@
 #
 
 set(FILES
```
Comment (Kadino) [inline Code/Framework/AzTest/AzTest/aztest_files.cmake:9]: I've convinced myself that UnitTest and TestTypes can move into AzTest, but I'll need an explanation on why all test modules will need the other files proposed in this move.

**Review by Kadino — CHANGES_REQUESTED** [review]

**Pull request closed without merging**